database/seeds/ModelSeeders/BeatmapSeeder.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+use Illuminate\Database\Seeder;
+
+class BeatmapSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        $beatmaps_array = [];
+        $beatmapset_array = [];
+        $overbeatmaps = [];
+        $overbeatmapsets = [];
+
+        $count_api_calls = 0;
+        $base_url = 'https://osu.ppy.sh/api/';
+        $api_key = env('OSU_API_KEY', null);
+        if (empty($api_key)) {
+            $this->command->error('Error: No OSU_API_KEY value set in .env file. Can\'t seed beatmap data!');
+
+            return;
+        }
+        $api = '&k='.$api_key;
+        $beatmaplimit = 20;
+
+        try {
+            $beatmaps = json_decode(file_get_contents($base_url.'get_beatmaps?since=2016-01-01%2000:00:00'.$api));
+            ++$count_api_calls;
+            $last_beatmapset = null;
+            $beatmap_diff_names = '';
+            $beatmapset_versions = 0;
+            $set_playcount = 0;
+
+            $first_map = true;
+
+            foreach ($beatmaps as $bm) {
+                // Here we are going to check if the current beatmap belongs to a new set, and make the set if necessary
+                if ($last_beatmapset === $bm->beatmapset_id || $first_map === true) {
+                    ++$beatmapset_versions;
+                    $beatmap_diff_names = $beatmap_diff_names.$bm->version.',';
+                    $set_playcount += $bm->playcount;
+                } else {
+                    // Create new beatmapset based on the PREVIOUS beatmap (since current one is in the next set)
```

Comment:
Yeah, seems you're right. Guess I'll need to write some extra code for the last beatmapset.